### DISCUSSION TITLE
Why does the .git file not come inside the zip?

### DISCUSSION THREAD

**Discussion by aus-it-admin** [community/community#30346] [Other Feature Feedback, Questions, & Ideas]
When I download as a zip, I'm usually handing code over to a client. Why do I need to clone the repo and then zip when there is a zip function in the browser that doesn't give me the complete repo? Is there a better way to do this?

**Comment by airtower-luna** [comment]
The ZIP download is supposed to give you only the contents of a repository at one specific commit, not a full Git repository. So no, there's no "download ZIP of Git repository" option.

> Is there a better way to do this?

How about giving the client access to the Git repository? Then they can clone themselves. In the end it depends on what the client actually needs.

**Reply by chriselswede** [reply]
And if it cannot be assumed that client has git?

**Comment by NikhilMath** [comment]
downloading the zip should work, that is interesting that when you do that you don't get the whole thing

**Comment by bndemerci** [comment]
This might be helpful for when someone is having trouble with lost **tokens** or **ssh keys**

**Reply by strangeprogrammer** [reply]
Which I am, though I guess I could always make some new SSH keys :/
